components/StorySeo/lib/getStoryExcerpt.tsx

```diff
@@ -0,0 +1,73 @@
+import { stringifyNode } from '@prezly/content-renderer-react-js';
+import type { ExtendedStory } from '@prezly/sdk';
+import { StoryFormatVersion } from '@prezly/sdk';
+import type { Node } from '@prezly/slate-types';
+import { isParagraphNode, isTextNode } from '@prezly/slate-types';
+
+function isNodeEmpty(node: Node): boolean {
+    if (isTextNode(node)) {
+        return !node.text.length;
+    }
+
+    return !node.children.length || (node.children as Node[]).every(isNodeEmpty);
+}
+
+function getNodeTextLength(node: Node): number {
+    if (isTextNode(node)) {
+        return node.text.length;
+    }
+
+    return (node.children as Node[]).reduce(
+        (total: number, child) => total + getNodeTextLength(child),
+        0,
+    );
+}
+
+const MAX_NODE_INDEX_FOR_TEXT_NODES = 5;
+const MAX_TOTAL_TEXT_LENGTH = 300;
+
+export function getStoryExcerpt(story: ExtendedStory): string {
```

Comment:
Works nice, good job 👍 

I am not sure if this should be part of the theme code 🤔 I think it would be better to have a function like this in one of the `@prezly/slate` packages. IMHO themes should not care about the story content structure - that's why we have the `@prezly/content-renderer-react-js`. Separation of concerns.

Also, now you'll have to duplicate this function in each theme unless you extract it into the theme kit, in which case you'll have to add `@prezly/content-renderer-react-js` and all the related packages as dependencies there.

The code for old editor stories can go to theme kit directly IMHO.

Reply:
That's a good point Lukas! This is exactly the question I asked in Linear :)

> Also, now you'll have to duplicate this function in each theme unless you extract it into the theme kit, in which case you'll have to add @prezly/content-renderer-react-js and all the related packages as dependencies there.

This one didn't come to my mind 🤔 
It is right, we don't want to bring extra deps to `theme-kit-nextjs` just for this method.

But I'm not entirely sure if Slate packages are a good place to store this function, too.
It's kinda in between the concerns of a theme and the Slate renderer.

Would be cool if @e1himself could give his opinion on that, because he's basically the owner of Slate repos right now.

Reply:
We decided that even better option would be to expose the `summary` from the API.